### ISSUE TITLE
Appium : Mobile Web: Safari: appium fails to switch back to default tab

### ISSUE THREAD

**Issue by Nishant-Taksande** [appium/appium#6971]
## The problem

Script Flow:
1. Launch a url in safari simulator: opens a base page with facebook login link
2. Click on Facebook link (opens a new safari window with facebook login)
3. Enter credentials and click on login.
4. Facebook window auto closes.
5. User is logged in to the app
6. Find an element on base page

Issue
Script works fine till step4 (no switching is required after step2)
Script goes in to an infinite loop at step 6
## Environment
- Appium version: 1.5.3
- Desktop OS/version: OS X 10.12, Xcode version 7.3.1
- Node.js version: v6.7.0
- Mobile platform/version: iOS 9.3
- Real device or emulator/simulator: Simulator
- Appium CLI or Appium.app|exe: CLI
## Details

Page1-AppHomePage
Page2-FacebookLoginPage

Analyzing the logs i see that.
After clicking the facebook link on Page1, appium is auto switching to page2
after logging to facebook on page2, page2 auto closes, there is a change in page1

appium detects this change, but does not switch to this page

[debug] [RemoteDebugger] Receiving data from remote debugger: '{"__argument":{"WIRApplicationIdentifierKey":"PID:11556","WIRListingKey":{"1":{"WIRTitleKey":"Buy Express | blablabla","WIRTypeKey":"WIRTypeWeb","WIRURLKey":"https://dev.blablabla.com/cart/so/nAMIoa","WIRPageIdentifierKey":1,"WIRConnectionIdentifierKey":"bd4b406e-2900-47f1-908e-478387643fe3"}}},"__selector":"_rpc_applicationSentListing:"}'
[debug] [RemoteDebugger] Page changed: {"1":{"WIRTitleKey":"Buy Express | blablabla","WIRTypeKey":"WIRTypeWeb","WIRURLKey":"https://dev.blablabla.com/cart/so/nAMIoa","WIRPageIdentifierKey":1,"WIRConnectionIdentifierKey":"bd4b406e-2900-47f1-908e-478387643fe3"}}
[debug] [iOS] Remote debugger notified us of a new page listing: {"appIdKey":"11556","pageArray":[{"id":1,"title":"Buy Express | Kachyng","url":"https://dev.blablabla.com/cart/so/nAMIoa","isKey":true}]}
[debug] [iOS] We are in the middle of selecting a page, ignoring
[HTTP] --> POST /wd/hub/session/7208b58e-8fd5-4bfd-93a7-442cbfd63563/element {"using":"xpath","value":"//div[@id='ka-head']//h5"}
[MJSONWP] Calling AppiumDriver.findElement() with args: ["xpath","//div[@id='ka-hea...
[debug] [iOS] Executing iOS command 'findElement'
## Link to Appium logs

Please find the logs at
https://gist.github.com/Nishant-Taksande/f769de4aa62ce3364015103d3e496710
## Code To Reproduce Issue

Please find the code here: 
If the base url does not work, Please update the code with any page having social sign in with facebook

https://gist.github.com/Nishant-Taksande/7d76fd5d2af4322b374f8f3cfbf7421c

**Comment by dpgraham** [comment]
Closing as duplicate of https://github.com/appium/appium/issues/6685
